src/d3a/models/strategy/update_frequency.py

```diff
@@ -209,7 +198,28 @@ def _post_bids(self, market, strategy):
             strategy.post_bid(market, bid.energy * self.get_updated_rate(market.time_slot),
                               bid.energy)
 
+    def update_market_cycle_bids(self, strategy):
+        # decrease energy rate for each market again, except for the newly created one
+        for market in strategy.area.all_markets[:-1]:
+            self.update_counter[market.time_slot] = 0
+            self._post_bids(market, strategy)
+
     def update_posted_bids_over_ticks(self, market, strategy):
         if self.time_for_price_update(strategy, market.time_slot):
             if strategy.are_bids_posted(market.id):
                 self._post_bids(market, strategy)
+
+
+class OffersUpdateFrequencyMixin(UpdateFrequencyMixin):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def update_market_cycle_offers(self, strategy):
```

Comment:
I think we can rename this to just `reset` because all this method does is to reset the update_counter and post updated bids / offers, so essentially to start from scratch. This happens during the market cycle event, that's why the name crept in this class.